### ISSUE TITLE
Cyrillic becomes questions �����

### ISSUE THREAD

**Issue by onimor** [dotnet/aspnetcore#37677]
.NET6 , VS22
[link to the project git](https://github.com/onimor/CyrillicTest)

Created a minimal api project. Added some Cyrillic
`app.MapGet("/", () => "Rus ?  фиг там был!");`
When I run it on windows, everything is fine.
I publish it in the docker hub, run it in the docker on windows - everything is fine. 
I buy a VPS (ubuntu), deploy this project in docker, get this instead of the Russian text: "��� ��� ���"

locale в ubuntu
```
LANG=ru_RU.UTF-8
LANGUAGE=
LC_CTYPE="ru_RU.UTF-8"
LC_NUMERIC="ru_RU.UTF-8"
LC_TIME="ru_RU.UTF-8"
LC_COLLATE="ru_RU.UTF-8"
LC_MONETARY="ru_RU.UTF-8"
LC_MESSAGES="ru_RU.UTF-8"
LC_PAPER="ru_RU.UTF-8"
LC_NAME="ru_RU.UTF-8"
LC_ADDRESS="ru_RU.UTF-8"
LC_TELEPHONE="ru_RU.UTF-8"
LC_MEASUREMENT="ru_RU.UTF-8"
LC_IDENTIFICATION="ru_RU.UTF-8"
LC_ALL=
```
locale docker

```
# locale
LANG=ru_RU.UTF-8
LANGUAGE=ru_RU:ru
LC_CTYPE="ru_RU.UTF-8"
LC_NUMERIC="ru_RU.UTF-8"
LC_TIME="ru_RU.UTF-8"
LC_COLLATE="ru_RU.UTF-8"
LC_MONETARY="ru_RU.UTF-8"
LC_MESSAGES="ru_RU.UTF-8"
LC_PAPER="ru_RU.UTF-8"
LC_NAME="ru_RU.UTF-8"
LC_ADDRESS="ru_RU.UTF-8"
LC_TELEPHONE="ru_RU.UTF-8"
LC_MEASUREMENT="ru_RU.UTF-8"
LC_IDENTIFICATION="ru_RU.UTF-8"
LC_ALL=ru_RU.UTF-8
```
I've been trying to fix it for a week. I don't understand what the problem is, why this behavior?

[Here's what I get](http://ovz3.j79805536.pxlzp.vps.myjino.ru/)

**Comment by pranavkm** [comment]
@onimor interestingly the raw file on Git shows similar issues with garbled text: https://raw.githubusercontent.com/onimor/CyrillicTest/master/CyrillicTest/Program.cs. Perhaps the file's encoding is stored using a windows specific encoding (e.g. https://en.wikipedia.org/wiki/Windows-1251)?

**Comment by onimor** [comment]
@pranavkm It's funny, but the file was really 1251 encoded, using notepad++ I made it utf8 and everything worked. 

To fix this for new files, I had to make this setting `Tools > Options > Environment > Documents > Save documents as Unicode when data cannot be saved in codepage` .  

And you know what's even funnier? in my Blazor WebAssembly project, the files are *.razor UTF8 , but the gRPC files are - 1251 and I see the Russian text on the site, but the gRPC data received become questions, now it's clear why this happens... But it is not clear why VS creates files in different encodings.
